### PR TITLE
feat: implement update operation for user service

### PR DESCRIPTION
**Pull request by yassir-ai**
Closes: #5

### CODE REVIEW

File: UserService/Controllers/UserController.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 using AutoMapper;
 using Microsoft.AspNetCore.Mvc;
 using UserService.Dtos;
+using UserService.Dtos;
```
Comment (acboujida) [inline UserService/Controllers/UserController.cs]: two identical lines

File: UserService/Controllers/UserController.cs [hunk 2]
```diff
@@ -71,4 +72,28 @@ public IActionResult GetUser([FromRoute]string id)
...
+    [HttpPut("{id}")]
+    public IActionResult UpdateUser([FromRoute] string id, [FromBody]UserDtoUpdate userUpdateDto)
+    {
+        if(!ModelState.IsValid) return BadRequest();
+
+        try
+        {
+            if(!_userRepository.UserExists(id)) return NotFound();
+
+            var user = _mapper.Map<User>(userUpdateDto);
```
Comment (acboujida) [inline UserService/Controllers/UserController.cs]: what happens to the other fields that don't exist in the userupdatedto? are they set to null in the user model? does that mean we could overwrite data unintentionally?

File: UserService/Controllers/UserController.cs [hunk 3]
```diff
@@ -71,4 +72,28 @@ public IActionResult GetUser([FromRoute]string id)
...
+
+            user.Id = id;
+
+            _userRepository.UpdateUser(user);
```
Comment (acboujida) [inline UserService/Controllers/UserController.cs:89]: maybe use the boolean returned by this method

**Review by acboujida — CHANGES_REQUESTED** [review]

File: UserService/Controllers/UserController.cs (hunk 1 above)
Comment (yassir-ai) [reply UserService/Controllers/UserController.cs]: I made a typing error, I'll fix it

File: UserService/Controllers/UserController.cs (hunk 3 above)
Comment (yassir-ai) [reply UserService/Controllers/UserController.cs:89]: We'll use it for wich purpose ?
It is more common in controller endpoints to return NoContent for PUT methods.

File: UserService/Controllers/UserController.cs (hunk 2 above)
Comment (yassir-ai) [reply UserService/Controllers/UserController.cs]: Native data types like int, float take 0 an 0,0 successively
Object like string, it takes null.
Maybe we should first track the last version of our object, so we can keep the values ​​of the fields unchanged

**Pull request merged**